src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -29,73 +30,127 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
+        }
+
+        protected override async Task<ImmutableArray<CodeAction>> GetRefactoringsForSingleParameterAsync(
+            Document document, IParameterSymbol parameter, SyntaxNode functionDeclaration, IMethodSymbol methodSymbol,
+            IBlockOperation blockStatementOpt, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            // Only should provide null-checks for reference types and nullable types.
+            if (!ParameterValidForNullCheck(document, parameter, semanticModel, blockStatementOpt, cancellationToken))
+            {
+                return ImmutableArray<CodeAction>.Empty;
             }
 
             // Great.  There was no null check.  Offer to add one.
             var result = ArrayBuilder<CodeAction>.GetInstance();
             result.Add(new MyCodeAction(
                 FeaturesResources.Add_null_check,
-                c => AddNullCheckAsync(document, parameter, functionDeclaration, method, blockStatementOpt, c)));
+                c => AddNullCheckAsync(document, parameter, functionDeclaration, methodSymbol, blockStatementOpt, c)));
 
             // Also, if this was a string, offer to add the special checks to 
             // string.IsNullOrEmpty and string.IsNullOrWhitespace.
             if (parameter.Type.SpecialType == SpecialType.System_String)
             {
                 result.Add(new MyCodeAction(
                     FeaturesResources.Add_string_IsNullOrEmpty_check,
-                    c => AddStringCheckAsync(document, parameter, functionDeclaration, method, blockStatementOpt, nameof(string.IsNullOrEmpty), c)));
+                    c => AddStringCheckAsync(document, parameter, functionDeclaration, methodSymbol, blockStatementOpt, nameof(string.IsNullOrEmpty), c)));
 
                 result.Add(new MyCodeAction(
                     FeaturesResources.Add_string_IsNullOrWhiteSpace_check,
-                    c => AddStringCheckAsync(document, parameter, functionDeclaration, method, blockStatementOpt, nameof(string.IsNullOrWhiteSpace), c)));
+                    c => AddStringCheckAsync(document, parameter, functionDeclaration, methodSymbol, blockStatementOpt, nameof(string.IsNullOrWhiteSpace), c)));
             }
 
             return result.ToImmutableAndFree();
         }
 
         protected abstract bool CanOffer(SyntaxNode body);
 
+        private async Task<Document> UpdateDocumentForRefactoringAsync(
+            Document document,
+            SyntaxNode functionDeclaration,
+            IBlockOperation blockStatementOpt,
+            List<int> listOfParametersOrdinals,
+            int position,
+            CancellationToken cancellationToken)
+        {
+            foreach (var index in listOfParametersOrdinals)
```

Comment:
please don't do this.  we don't need to loop N times over making N documents and having to go *re-find* parameters after making edits.  this is a very very brittle way to do things and should def be avoided.

Reply:
Will try to address this in a follow-up PR